bundle/deploy/files/upload.go

```diff
@@ -28,10 +37,12 @@ func (m *upload) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 		return diag.FromErr(err)
 	}
 
+	sync.Close()
+
```

Comment:
The call to `New()` will spawn the routine, so `Close()` must always be called, or we may leak the routine.

You can call `defer sync.Close()` on L35 to make it always close regardless of the next step erroring or not.